File: front-end/ShapeFactory.js
import Konva from 'konva';

class ShapeFactory {
  static createShape(type, options) {
    switch (type) {
      case 'line':
        return new Konva.Line(options)  
      case 'circle':
        return new Konva.Circle(options);
      case 'rectangle':
        return new Konva.Rect(options);
      case 'ellipse':
        return new Konva.Ellipse(options);
      case 'triangle':
        return new Konva.RegularPolygon({
          ...options,
          sides: 3,
        });
    }
  }
}

export default ShapeFactory;

File: BackendSrc/src/main/java/com/jetbrains/csed/paint/Shapes/ShapeFactory.java
package com.jetbrains.csed.paint.Shapes;

import com.jetbrains.csed.paint.Shapes.*;
import com.jetbrains.csed.paint.Shapes.Rectangle;

public class ShapeFactory {
    public com.jetbrains.csed.paint.Shapes.Shape getShape(ShapeDTO data) {
        if (data == null) {
            return null;
        }
        else if (data.shapeType.equalsIgnoreCase("circle")) {
            return new Circle(data);
        }
        else if (data.shapeType.equalsIgnoreCase("rectangle")) {
            return new Rectangle(data);
        }
        else if (data.shapeType.equalsIgnoreCase("triangle")) {
            return new Triangle(data);
        }
        else if (data.shapeType.equalsIgnoreCase("line_segment")) {
            return new LineSegment(data);
        }
        else if (data.shapeType.equalsIgnoreCase("ellipse")) {
            return new Ellipse(data);
        }
        else if (data.shapeType.equalsIgnoreCase("square")) {
            return new Square(data);
        }
        return null;
    }
}
